lib/Migrator/APIDiffMigratorPass.cpp

```diff
@@ -327,17 +352,76 @@ struct APIDiffMigratorPass : public ASTMigratorPass, public SourceEntityWalker {
         Walker.walk(FuncRefContainer);
         Editor.replace(Walker.Result, View.base());
       }
-      unsigned Idx = 0;
-      for (auto LR :getCallArgLabelRanges(SM, Arg,
-                                          LabelRangeEndAt::LabelNameOnly)) {
-        if (Idx < View.argSize()) {
-          auto Label = View.args()[Idx++];
-
-          // FIXME: We update only when args are consistently valid.
-          if (Label != "_" && LR.getByteLength())
-            Editor.replace(LR, Label);
-        }
+      emitRenameLabelChanges(Arg, View, {});
+    }
+  }
+
+  bool handleTypeHoist(ValueDecl *FD, CallExpr* Call, Expr *Arg) {
+    TypeMemberDiffItem *Item = nullptr;
+    for (auto *I: getRelatedDiffItems(FD)) {
+      Item = dyn_cast<TypeMemberDiffItem>(I);
+      if (Item)
+        break;
+    }
+    if (!Item)
+      return false;
+    if (Item->Subkind == TypeMemberDiffItemSubKind::SimpleReplacement)
+      return false;
+    if (*Item->selfIndex)
+      return false;
+    std::vector<CallArgInfo> AllArgs =
+      getCallArgInfo(SM, Arg, LabelRangeEndAt::LabelNameOnly);
+    if (!AllArgs.size())
+      return false;
+    DeclNameViewer NewName = Item->getNewName();
+    llvm::SmallVector<unsigned, 2> IgnoredArgIndices;
+    IgnoredArgIndices.push_back(*Item->selfIndex);
+    if (auto RI = Item->removedIndex)
+      IgnoredArgIndices.push_back(*RI);
+    emitRenameLabelChanges(Arg, NewName, IgnoredArgIndices);
+    auto *SelfExpr = AllArgs[0].ArgExp;
```

Comment:
good catch, line 370 will return if SelfIndex is not 0. Current transformation mechanism may have limitation to handle non-zero selfindex. Fortunately, we don't see any of them in real frameworks.